java/ql/src/semmle/code/java/security/UrlRedirect.qll

```diff
@@ -1,12 +1,14 @@
-import java
-import semmle.code.java.frameworks.Servlets
+/** Provides classes to reason about URL redirect attacks. */
+
 import semmle.code.java.dataflow.DataFlow
+import semmle.code.java.frameworks.Servlets
+
+/** A URL redirection sink */
+abstract class UrlRedirectSink extends DataFlow::Node { }
 
-/**
- * A URL redirection sink.
- */
-class UrlRedirectSink extends DataFlow::ExprNode {
-  UrlRedirectSink() {
+/** A Servlet URL redirection sink. */
+class ServletUrlRedirectSink extends UrlRedirectSink {
```

Comment:
This class should be `private`.

Reply:
Yes, just realised this when moving sources and sinks for another PR. Will update asap.